src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1046,11 +1047,28 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
             }
         }
 
+        private static SyntaxKind[] _lazyContextualKeywordKindsArray;
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.WhenKeyword; i++)
+            if (_lazyContextualKeywordKindsArray == null)
             {
-                yield return (SyntaxKind)i;
+                ushort firstKeywordIdInclusive = (ushort)SyntaxKind.YieldKeyword;
```

Comment:
I believe you can actually loop over `SyntaxKind` values though. :smile: I prefer to use more restricted types where possible, but it's not a strong/enforced preference.